Plugins doesn't work on Gradle 5.2 after updating to 3.0.0

```
Gradle : 5.2
ktlintPlugin = "7.0.0"
ktlint = "0.30.0"
kotlin = "1.3.20"
```

The plugin is not working after upgrading to Gradle `5.2` and ktlint-plugin to `3.0.0`

```
* What went wrong:
An exception occurred applying plugin request [id: 'org.jlleitschuh.gradle.ktlint', version: '7.0.0']
> Failed to apply plugin [id 'org.jlleitschuh.gradle.ktlint']
   > Could not create plugin of type 'KtlintPlugin'.
      > Could not generate a decorated class for class org.jlleitschuh.gradle.ktlint.KtlintPlugin.
         > com/android/build/gradle/BaseExtension
```

Stacktrace
```
Caused by: java.lang.NoClassDefFoundError: com/android/build/gradle/BaseExtension
        at org.gradle.internal.reflect.ClassInspector.inspectClass(ClassInspector.java:69)
        at org.gradle.internal.reflect.ClassInspector.visitGraph(ClassInspector.java:55)
        at org.gradle.internal.reflect.ClassInspector.inspect(ClassInspector.java:35)
        at org.gradle.internal.instantiation.AbstractClassGenerator.inspectType(AbstractClassGenerator.java:199)
        at org.gradle.internal.instantiation.AbstractClassGenerator.generateUnderLock(AbstractClassGenerator.java:152)
        ... 162 more
Caused by: java.lang.ClassNotFoundException: com.android.build.gradle.BaseExtension
        ... 167 more```

--- a/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt
+++ b/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintPlugin.kt
@@ -197,27 +197,6 @@ open class KtlintPlugin : Plugin<Project> {
         }
     }
 
-    private fun BaseExtension.addVariantsMetaTasks(
-        target: Project,
-        multiplatformTargetName: String? = null
-    ) {
-        variants?.all { variant ->
-            val variantCheckTask = target.createAndroidVariantMetaKtlintCheckTask(
-                variant.name,
-                multiplatformTargetName
-            )
-            val variantFormatTask = target.createAndroidVariantMetaKtlintFormatTask(
-                variant.name,
-                multiplatformTargetName
-            )
-            variant.sourceSets.forEach { sourceSet ->
-                val sourceSetName = "${multiplatformTargetName?.capitalize() ?: ""}${sourceSet.name.capitalize()}"
-                variantCheckTask.configure { it.dependsOn(sourceSetName.sourceSetCheckTaskName()) }
-                variantFormatTask.configure { it.dependsOn(sourceSetName.sourceSetFormatTaskName()) }
-            }
-        }
-    }
-
     private fun PluginHolder.applyKtLintKonanNative(): (Plugin<in Any>) -> Unit {
         return {
             val ktLintConfig = createConfiguration(target, extension)
@@ -345,23 +324,6 @@ open class KtlintPlugin : Plugin<Project> {
         reporters.set(extension.reporters)
     }
 
-    private fun Project.createAndroidVariantMetaKtlintCheckTask(
-        variantName: String,
-        multiplatformTargetName: String? = null
-    ): TaskProvider<Task> = registerTask(variantName.androidVariantMetaCheckTaskName(multiplatformTargetName)) {
-        group = VERIFICATION_GROUP
-        description = "Runs ktlint on all kotlin sources for android $variantName variant in this project."
-    }
-
-    private fun Project.createAndroidVariantMetaKtlintFormatTask(
-        variantName: String,
-        multiplatformTargetName: String? = null
-    ): TaskProvider<Task> = registerTask(variantName.androidVariantMetaFormatTaskName(multiplatformTargetName)) {
-        group = FORMATTING_GROUP
-        description = "Runs ktlint formatter on all kotlin sources for android $variantName" +
-            " variant in this project."
-    }
-
     private fun setCheckTaskDependsOnKtlintCheckTask(
         project: Project,
         ktlintCheck: TaskProvider<KtlintCheckTask>
@@ -406,4 +368,47 @@ open class KtlintPlugin : Plugin<Project> {
             }
         }
     }
+}
+
+/**
+ * This is not scoped inside of [KtlintPlugin] because of these issues:
+ *  - https://github.com/JLLeitschuh/ktlint-gradle/issues/201
+ *  - https://github.com/gradle/gradle/issues/8411
+ */
+private fun BaseExtension.addVariantsMetaTasks(
+    target: Project,
+    multiplatformTargetName: String? = null
+) {
+    variants?.all { variant ->
+        val variantCheckTask = target.createAndroidVariantMetaKtlintCheckTask(
+            variant.name,
+            multiplatformTargetName
+        )
+        val variantFormatTask = target.createAndroidVariantMetaKtlintFormatTask(
+            variant.name,
+            multiplatformTargetName
+        )
+        variant.sourceSets.forEach { sourceSet ->
+            val sourceSetName = "${multiplatformTargetName?.capitalize() ?: ""}${sourceSet.name.capitalize()}"
+            variantCheckTask.configure { it.dependsOn(sourceSetName.sourceSetCheckTaskName()) }
+            variantFormatTask.configure { it.dependsOn(sourceSetName.sourceSetFormatTaskName()) }
+        }
+    }
+}
+
+private fun Project.createAndroidVariantMetaKtlintCheckTask(
+    variantName: String,
+    multiplatformTargetName: String? = null
+): TaskProvider<Task> = registerTask(variantName.androidVariantMetaCheckTaskName(multiplatformTargetName)) {
+    group = VERIFICATION_GROUP
+    description = "Runs ktlint on all kotlin sources for android $variantName variant in this project."
+}
+
+private fun Project.createAndroidVariantMetaKtlintFormatTask(
+    variantName: String,
+    multiplatformTargetName: String? = null
+): TaskProvider<Task> = registerTask(variantName.androidVariantMetaFormatTaskName(multiplatformTargetName)) {
+    group = FORMATTING_GROUP
+    description = "Runs ktlint formatter on all kotlin sources for android $variantName" +
+        " variant in this project."
 }

